lib/rucio/extensions/forecast.py

```diff
@@ -0,0 +1,195 @@
+                attribs = list_rse_attributes(rse['rse'])
+                try:
+                    self._rse2site[rse['rse']] = attribs['site']
+                    self._rseid2site[rse['id']] = attribs['site']
+                except KeyError:
+                    logging.warning('(T3CModel rse2site mapper) No site for ' + rse['rse'])
+                    continue
+                if attribs['site'] not in self._site2rses.keys():
+                    self._site2rses[attribs['site']] = []
+                    self._site2rseids[attribs['site']] = []
+                self._site2rses[attribs['site']].append(rse['rse'])
+                self._site2rseids[attribs['site']].append(rse['id'])
+
+    # Site to RSE and RSE to site conversions
+    def rse2site(self, rse_name):
+        """
+        Translate RSE to Site name using Rucio DB  mapping.
+        """
+        try:
+            return self._rse2site[rse_name]
+        except KeyError:
+            return ''
+
+    def site2rses(self, site):
+        """
+        Translate Site name to all the RSEs in this site.
+        """
+        try:
+            return self._site2rses[site]
+        except KeyError:
+            return []
+
+    # Site to RSE and RSE to site conversions
+    def rseid2site(self, rse_id):
+        """
+        Translate RSE to Site name using Rucio DB  mapping.
+        """
+        try:
+            return self._rseid2site[rse_id]
+        except KeyError:
+            return ''
+
+    def site2rseids(self, site):
+        """
+        Translate Site name to all the RSEs in this site.
+        """
+        try:
+            return self._site2rseids[site]
+        except KeyError:
+            return []
+
+    # Net time prediction
+    def recover_params(self, link):
+        """
+        Get the parameters (rate, overhead, diskrw) to calculate
+        rate_pred = (size / (size / rate) + overhead) < diskrw
+        used to calculate the network transfer estimation
+        """
+        try:
+            values = self.model[link]
+        except KeyError:
+            link = 'STANDARD_LINK'
+            values = self.model['STANDARD_LINK']
+        return link, values['rate'], values['overhead'], values['diskrw']
+
+    def predict_n(self, link, size):
+        """
+        Use the parameters for the model return the network transfert estimation
+        for a given link.
+        returns: Number of seconds the transfer is going to take
+        """
+        link_true, rate, overhead, diskrw = self.recover_params(link)
+        if link_true == 'STANDARD_LINK':
+            logging.warning('W: Link ' + link + ' not found, using standard parameters...')
+        else:
+            logging.info('I: Using ' + link + ' data from model to predict.')
+        rate_pred = min((size / (size / rate) + overhead), diskrw)
+        return size / rate_pred
+
+    # Queue time prediction
+    @read_session
+    def get_submitted_at_to_rucio(self, src, dst, act, session=None):
```

Comment:
Hi JJ, I know it is a bit pedantic, but please try to do the docstrings similar to other files with :param: descriptions of each parameter and, if applicable, also a :return: description.

Reply:
Not at all. My bad. I will fix it ASAP.